### PR TITLE
Check if SELinux is installed on system

### LINKED ISSUE #21622: Selinux module fails if selinux is not installed
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
selinux module

##### ANSIBLE VERSION

```
ansible 2.2.1.0
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY
If a task wants to set selinux to `permissive`, but selinux is not installed (RPMS have been erased, /etc/selinux/config doesn't exist) the module fails because it cannot find the config file. This particular case should be OK, as setting it to permissive is effectively disabling it, and it's already disabled.

##### STEPS TO REPRODUCE
deploy a CentOS/RHEL vm, delete the selinux rpms, rm -rf /etc/selinux
Then attempt to disable to set selinux to permissive

```
- selinux:
    state: permissive

# Disable SELinux
- selinux:
    state: disabled

```

##### EXPECTED RESULTS

Since selinux is not installed, it should be a noop

##### ACTUAL RESULTS

    ```
    IOError: [Errno 2] No such file or directory: '/etc/selinux/config'\n",
    "module_stdout": "", "msg": "MODULE FAILURE"}
    ```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* Refactor code
* PEP8 fixes
* Check if /etc/selinux/config file exists before
  proceeding any other operations

Fixes #21622

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/system/selinux.py
test/sanity/pep8/legacy-files.txt

##### ANSIBLE VERSION
```
2.4 devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
ready_for_review

File: lib/ansible/modules/system/selinux.py [hunk 1]
```diff
@@ -71,76 +71,82 @@
...
 except ImportError:
     HAS_SELINUX = False
+from ansible.module_utils.basic import AnsibleModule
+
+
+def read_config(config_file):
+    with open(config_file, 'r') as read_file:
+        lines = read_file.readlines()
+    return lines
+
```
Comment (bcoca) [inline lib/ansible/modules/system/selinux.py]: you could just use `get_file_content` from module_utils/facts.py

Comment (Akasurde) [reply lib/ansible/modules/system/selinux.py]: Done. I used `get_file_lines`

**Comment by Akasurde** [comment]
ready_for_review

**Comment by gundalow** [comment]
@goozbach I believe you are the maintainer of the SELinux module, could you please review this PR.

File: lib/ansible/modules/system/selinux.py [hunk 2]
```diff
@@ -149,79 +150,81 @@ def main():
...
-    changed=False
-    msgs                  = []
-    configfile            = module.params['configfile']
-    policy                = module.params['policy']
-    state                 = module.params['state']
-    runtime_enabled       = selinux.is_selinux_enabled()
-    runtime_policy        = selinux.selinux_getpolicytype()[1]
-    runtime_state         = 'disabled'
-    if (runtime_enabled):
+    changed = False
```
Comment (goozbach) [inline lib/ansible/modules/system/selinux.py:153]: not a fan of the aligned assignments (is that a pep8 thing? I'm not really a python guy)

**Comment by goozbach** [comment]
Been a long time since I've touched this, however everything looks to be on the up-and-up.

I'll run an integration test tomorrow, if you don't hear from me in 24hr consider the state "shipit"

File: lib/ansible/modules/system/selinux.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/system/selinux.py:153]: @goozbach Yes. It is to match Ansible module standards.

**Comment by Akasurde** [comment]
shipit

**Comment by Akasurde** [comment]
bump for review

**Comment by Akasurde** [comment]
ready_for_review

**Comment by Akasurde** [comment]
@gundalow @goozbach @bcoca Thanks for review and comments.

**Pull request merged**